pkg/strategy/common/profit_fixer.go

```diff
@@ -208,6 +220,49 @@ func buildTokenFeeDatePrices(ctx context.Context, sessions map[string]types.Exch
 			return nil, err
 		}
 	}
+	// USD prices
+	// only build USD prices if there are trades with USD* fee currency
+	// ex: USDT, USDC
+	for exName := range usdFeeExchanges {
+		ex, ok := exchanges[exName]
+		if !ok {
+			continue
+		}
+		markets, ok := markets[exName]
+		if !ok {
+			continue
+		}
+		quoteMarket, ok := markets[quoteCurrency+"USD"]
+		if !ok {
+			continue
+		}
+		if err := func() error {
+			query := &batch.KLineBatchQuery{Exchange: ex}
+			kLineC, errC := query.Query(ctx, quoteMarket.Symbol, types.Interval1d, startTime, endTime)
+			for {
+				select {
+				case <-ctx.Done():
+					return ctx.Err()
+				case kline, ok := <-kLineC:
+					if !ok {
+						err := <-errC
+						return err
+					}
+					// we still use the last day's closing price to calculate the USD fee price.
+					// This is different from the actual fee calculation while trading which use the close price of the last minute.
+					// But assuming the price of USD stablecoins do not fluctuate much within a day,
+					// this should be acceptable.
+					tokenFeePrices[tokenFeeKey{
+						token:        "USD",
+						exchangeName: ex.Name(),
+						date:         kline.StartTime.Time().Add(24 * time.Hour).Format(time.DateOnly),
+					}] = fixedpoint.One.Div(kline.Close)
```

Comment:
You need to detect the conversion side here? eg

USDT/TWD and USDT/USD

sometimes it could be USDC/USDT